crates/crypto/core/src/lib.rs

```diff
@@ -130,3 +130,91 @@ pub trait KeyType: Sized + 'static {
     ) -> Result<Self::Signature, Self::Error>;
     fn verify(public: &Self::Public, msg: &[u8], signature: &Self::Signature) -> bool;
 }
+
+#[macro_export]
+macro_rules! impl_crypto_tests {
+    ($ecdsa_type:ty, $signing_key:ty, $signature:ty) => {
```

Comment:
Please rename `ecdsa_type` to `key_type` since it seems you're using not only ecdsa for this.